app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt

```diff
@@ -0,0 +1,338 @@
+import androidx.compose.material.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.Favicon
+import org.mozilla.fenix.compose.PrimaryText
+import org.mozilla.fenix.compose.SecondaryText
+import org.mozilla.fenix.theme.FirefoxTheme
+
+private val LIST_ITEM_HEIGHT = 56.dp
+
+private val ICON_SIZE = 24.dp
+
+/**
+ * List item used to display a label with an optional description text and
+ * an optional, interactable icon at the end.
+ *
+ * @param label The label in the list item.
+ * @param description An optional description text below the label.
+ * @param onClick Called when the user clicks on the item.
+ * @param icon Optional [DrawableRes] used to display a drawable at the end of the list item.
+ * @param iconDescription Content description of the icon.
+ * @param onIconClick Called when the user clicks on the icon.
+ */
+@Composable
+fun TextListItem(
+    label: String,
+    description: String? = null,
+    onClick: (() -> Unit)? = null,
+    @DrawableRes icon: Int? = null,
+    iconDescription: String? = null,
+    onIconClick: (() -> Unit)? = null,
+) {
+    ListItem(
+        label = label,
+        description = description,
+        onClick = onClick,
+        afterListAction = {
+            icon?.let {
+                ListItemIcon(
+                    icon = icon,
+                    iconDescription = iconDescription,
+                    onClick = onIconClick,
+                )
+            }
+        }
+    )
+}
+
+/**
+ * List item used to display a label and a [Favicon] with an optional description text and
+ * an optional, interactable icon at the end.
+ *
+ * @param label The label in the list item.
+ * @param description An optional description text below the label.
+ * @param onClick Called when the user clicks on the item.
+ * @param url Website [url] for which the favicon will be shown.
+ * @param icon Optional [DrawableRes] used to display a drawable at the end of the list item.
+ * @param iconDescription Content description of the icon.
+ * @param onIconClick Called when the user clicks on the icon.
+ */
+@Composable
+fun FaviconListItem(
+    label: String,
+    description: String? = null,
+    onClick: (() -> Unit)? = null,
+    url: String,
+    @DrawableRes icon: Int? = null,
+    iconDescription: String? = null,
+    onIconClick: (() -> Unit)? = null,
+) {
+    ListItem(
+        label = label,
+        description = description,
+        onClick = onClick,
+        beforeListAction = {
+            Spacer(modifier = Modifier.width(16.dp))
+
+            Favicon(
+                url = url,
+                size = ICON_SIZE,
+            )
+
+            Spacer(modifier = Modifier.width(16.dp))
```

Comment:
This might be a good instance where we should be adding a Modifier parameter to `Favicon`. and prefer to add a `Modifier.padding(horizontal = 16.dp)` over `Spacer`. Also, see https://chris.banes.dev/always-provide-a-modifier/

Reply:
I did find it odd that `Favicon` wasn't written to have a `Modifier` as a param. I'll make a follow-up story for this.